relayer/src/foreign_client.rs

```diff
@@ -45,6 +45,13 @@ const MAX_MISBEHAVIOUR_CHECK_DURATION: Duration = Duration::from_secs(120);
 
 const MAX_RETRIES: usize = 5;
 
+/// Optional limit on the amount of results to fetch when querying
+/// the client consensus states.
+/// Rationale: The consensus state query is one of the most expensive
```

Comment:
What happens then when this optimistic query fails, i.e., 10 results turned out to not be enough?

Reply:
be4e626f